Electron/src/renderer/App.tsx

```diff
@@ -63,106 +64,108 @@ function App() {
       {!isLogged && !isSigningUp && (
         <StartMenu setIsLogged={setIsLogged} setIsSigningUp={setIsSigningUp} />
       )}
-      {isLogged && (
-        <div className={`App d-flex flex-column ${styles.appBackground}`}>
-          <StickyHeader handleLogout={handleLogout} />
-
-          <div className="d-flex">
-            <Sidebar
-              refreshSidebarTriggerValue={refreshSidebarTriggerValue}
-              refreshSidebarData={refreshSidebarData}
-            />
-            <div
-              className={`App d-flex container-fluid ${styles.mainContentWrapper}`}
-            >
-              <Routes>
-                <Route
-                  path="/playlist/:id"
-                  element=<Playlist
-                    changeSongName={changeSongName}
-                    refreshSidebarData={refreshSidebarData}
+      <SongNameChangeContextProvider>
+        {isLogged && (
+          <div className={`App d-flex flex-column ${styles.appBackground}`}>
+            <StickyHeader handleLogout={handleLogout} />
+
+            <div className="d-flex">
+              <Sidebar
+                refreshSidebarTriggerValue={refreshSidebarTriggerValue}
+                refreshSidebarData={refreshSidebarData}
+              />
+              <div
+                className={`App d-flex container-fluid ${styles.mainContentWrapper}`}
+              >
+                <Routes>
+                  <Route
+                    path="/playlist/:id"
+                    element=<Playlist
+                      changeSongName={changeSongName}
```

Comment:
`changeSongName` is still being passed as argument here, we should check for the function usages and use the context instead